compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/ComposeContainer.desktop.kt

```diff
@@ -139,6 +139,28 @@ internal class ComposeContainer(
         composeSceneFactory = ::createComposeScene,
     )
 
+    /**
+     * The listener to [windowContainer] size and position changes.
+     */
+    private val windowContainerComponentListener = object : ComponentAdapter() {
+        override fun componentResized(e: ComponentEvent?) = onWindowContainerSizeChanged()
+        override fun componentMoved(e: ComponentEvent?) = onWindowContainerPositionChanged()
+    }
+
+    /**
+     * The listener to [window] size changes.
+     */
+    private val windowSizeListener = object : ComponentAdapter() {
+        override fun componentResized(e: ComponentEvent?) {
+            if (!container.isDisplayable) return
+
+            // When the window is moved to a display with a different density, componentResized is
+            // called on the window, but not on `windowContainer`, so we need to update the size
+            // here too.
+            windowContext.setContainerSize(windowContainer.sizeInPx)
```

Comment:
Please call `onWindowContainerSizeChanged` instead - layers should be updated too (for dialog centering for example).
Please recreck your fix with all layer types (canvas, component, window)